### DISCUSSION TITLE
Server-side characters not saving correctly when players leave the server

### DISCUSSION THREAD

**Discussion by DaMarine** [Pryaxis/TShock#1878] [Q&A]
As described in the title, some players on my server are losing some, or all, of their inventory when disconnecting from the server and reconnecting after it has restarted. We are using server-side characters, but the server is not being shut down forcefully nor are the players disconnecting using any unusual means. So far we have experienced players losing their entire inventory, or everything apart from the hotbar, and this occurs with multiple members per restart. The save interval has been tested at 5 and 1 minutes, respectfully, with this occurring at both intervals.

**Comment by Olink** [comment]
Can you please include steps to reproduce?

I assume a player joined your server and performed the following:
```
>joined the server
/register name password
/login name password
>played the game
>left the server
>joined the server
/login name password
>observed missing inventory.
```

**Comment by DaMarine** [comment]
Yep, those are the exact steps we all went through
